[Term Entry] C++ Math Functions: `trunc()` (#1036)
Codecademy/docs: content/cpp/concepts/math-functions/terms/trunc/trunc.md

---
Title: 'trunc()' 
Description: 'Returns the integral part of the argument.' 
Subjects:
  - 'Computer Science'
Tags: 
  - 'Functions'
  - 'Arithmetic'
CatalogContent:
  - 'learn-c-plus-plus'
  - 'paths/computer-science'
---

> The **`trunc()`** function returns the integral part of the argument as a value of [type](https://www.codecademy.com/resources/docs/cpp/data-types) `double`, `float`, or `long double`. The fractional part is truncated, not rounded.

Reviewer: ```suggestion
The **`trunc()`** function rounds the specified argument towards zero and returns the nearest integral value not larger in magnitude than the argument. Its return value can be of [types](https://www.codecademy.com/resources/docs/cpp/data-types) `double`, `float`, or `long double`. 
```

> The **`trunc()`** function rounds the specified argument towards zero and returns the nearest integral value not larger in magnitude than the argument. Its return value can be of [types](https://www.codecademy.com/resources/docs/cpp/data-types) `double`, `float`, or `long double`. 

Reviewer 2: ```suggestion
The **`trunc()`** function rounds the specified argument towards zero and returns the nearest integral value not larger in magnitude than the argument. The [data type](https://www.codecademy.com/resources/docs/cpp/data-types) of the return value can be `double`, `float`, or `long double`. 
```

## Syntax

> ```cpp

Reviewer 2: ```suggestion
```pseudo
```

std::trunc(x);
```

> 

Reviewer 2: ```suggestion

The `x` parameter is the value to be truncated.

```

## Example

```cpp
#include <iostream>
#include <cmath>

> int main() {
>     double x = -3.14159;
>     double result;
> 
>     result = std::trunc(x);
> 
>     std::cout << "trunc(" << x << ") = " << result << "\n";
>     // Output: trunc(3.14159) = -3
> }

Reviewer 2: ```suggestion
int main() {
  double x = -3.14159;
  double result;

  result = std::trunc(x);

  std::cout << "trunc(" << x << ") = " << result << "\n";
}
```

```

> 

Reviewer 2: ```suggestion

This will produce the following output:

```

And then add a newline, followed by a \```shell code block that looks like this:

```shell
// Output: trunc(3.14159) = -3
```

## Codebyte Example

> Use `trunc()`  to strip off the decimal part of `12.3456` and return `12`.

Reviewer: ```suggestion
The following example uses the `trunc()` function to round `12.3456` towards zero returning the value `12`:
```

```codebyte/cpp
#include <iostream>
#include <cmath>

> int main() {
>     double n = 12.3456;
>     double result;
> 
>     result = std::trunc(n);
> 
>     std::cout << "The result is " << result << "!\n";
> }

Reviewer 2: ```suggestion
int main() {
  double n = 12.3456;
  double result;

  result = std::trunc(n);

  std::cout << "The result is " << result << "!\n";
}
```

---
Title: 'trunc()' 
Description: 'Returns the integral part of the argument.' 
Subjects:
  - 'Computer Science'
Tags: 
  - 'Functions'
  - 'Arithmetic'
CatalogContent:
  - 'learn-c-plus-plus'
  - 'paths/computer-science'
---

> The **`trunc()`** function returns the integral part of the argument. It rounds the argument towards zero and returns the nearest integer that is not greater in magnitude than the argument.

Reviewer 3: The second sentence should be rewritten to take into account negative numbers. If a negative number is rounded toward zero, it invalidates the second clause because the result would be greater than the argument.

## Syntax

```cpp
std::trunc(x);
```

## Example

```cpp
#include <iostream>
#include <cmath>

int main() {
    double x = 3.14159;
    double result;

    result = std::trunc(x);

    std::cout << "trunc(" << x << ") = " << result << "\n";
    // Output: trunc(3.14159) = 3
}
```

## Codebyte Example

> Use `trunc()` to round the double `12.3456` down to the nearest integer by removing the decimal part:

Reviewer 3: The phrasing here is misleading since the `trunc()` function doesn't round to the nearest integer (though it does in this case) but strips off the bit after the decimal. (i.e. if the argument was 12.9, the result would still be 12.)

---
Title: 'trunc()' 
Description: 'Returns the integral part of the argument.' 
Subjects:
  - 'Computer Science'
Tags: 
  - 'Functions'
  - 'Arithmetic'
CatalogContent:
  - 'learn-c-plus-plus'
  - 'paths/computer-science'
---

> The **`trunc()`** function rounds the specified argument towards zero and returns the nearest integral value not larger in magnitude than the argument. The [data type](https://www.codecademy.com/resources/docs/cpp/data-types) of the return value can be `double`, `float`, or `long double`.

Reviewer 2: ```suggestion
The **`trunc()`** function rounds the specified argument towards zero and returns the nearest integral value not larger in magnitude than the argument.
```

## Syntax

```pseudo
std::trunc(x);
```

> The `x` parameter is the value to be truncated.

Reviewer 2: ```suggestion
The `x` parameter is the value to be truncated.

The [data type](https://www.codecademy.com/resources/docs/cpp/data-types) of the return value can be `double`, `float`, or `long double`.
```

Reviewer 2: @sudojarvis Looks like this was resolved without the change being implemented. We should include this text as I think its important to describe the return type(s) in the "Syntax" section.

## Example

```cpp
#include <iostream>
#include <cmath>

int main() {
  double x = -3.14159;
  double result;

  result = std::trunc(x);

  std::cout << "trunc(" << x << ") = " << result << "\n";
}
```

> This will produce the following output:
> 
> This will produce the following output:

Reviewer 2: ```suggestion
This will produce the following output:
```

>   ```pseudo

Reviewer 2: ```suggestion
  ```shell
```

>   Output: trunc(-3.14159) = -3

Reviewer 2: ```suggestion
trunc(-3.14159) = -3
```
